encrypted-config-value/src/main/java/com/palantir/config/crypto/value/RsaEncryptedValue.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.config.crypto.value;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableRsaEncryptedValue.class)
+@JsonDeserialize(as = ImmutableRsaEncryptedValue.class)
+@JsonTypeName("RSA")
+public abstract class RsaEncryptedValue extends EncryptedValue {
```

Comment:
Happy to do this as a separate/follow-up PR, but just as `AES-GCM` has specific configuration values, `RSA-OAEP-MGF1` has configuration values as well -- specifically, the hash function for `OAEP` and `MGF1` should both be separately configurable as either `SHA-1` or `SHA-256`. The legacy values use `SHA-256` for `OAEP` and `SHA-1` for `MGF1`, but we may also want to change the "new" values to be the more standard `SHA-256` for both (in which case we will also need to update the algorithm class).

See https://github.com/palantir/encrypted-config-value/issues/58#issuecomment-294089060 for details.